src/aics-image-viewer/shared/utils/urlParsing.ts

```diff
@@ -272,14 +268,31 @@ class DataParams {
    * See `ManifestJson` for the type definition.
    */
   manifest?: string = undefined;
+  /** The name of a dataset in the Cell Feature Explorer database. Used with `id`. */
+  dataset?: string = undefined;
+  /** The ID of a cell within the loaded dataset. Used with `dataset`. */
+  id?: string = undefined;
+  /** The key of a collection of scenes stored in local storage. Overrides `url` unless `msgorigin` is present. */
+  storageid?: string = undefined;
   /**
-   * The name of a dataset in the Cell Feature Explorer database. Used with `id`.
+   * The origin of an opening window that wants to send a message to this window.
+   *
+   * The presence of both this param and `storageid` implies that this window has just been opened by another app, and
+   * that the opening app has more data to send. Until that message is received, we fall back to `url`. Once that
+   * message arrives, the scenes to open are written to local storage at key `storageid` and `msgorigin` is removed,
+   * allowing the window to switch to reading local storage.
+   *
+   * All this happens independently of URL parsing, so the only meaningful thing this parsing code does with this param
+   * is check whether it is present.
    */
-  dataset?: string = undefined;
+  msgorigin?: string = undefined;
   /**
-   * The ID of a cell within the loaded dataset. Used with `dataset`.
+   * If this param is present, the opening window wants to send more scene URLs via `postMessage`.
+   *
+   * The value of this param should be the total number of scenes the opening window wants to send, so we can set up
+   * the app with the correct number of scenes before knowing what all the scenes are.
    */
-  id?: string = undefined;
+  msgscenes?: string = undefined;
```

Comment:
I thought we didn't actually need this.  Can you add to the comment about why this has to be sent separately from the actual data?